pkgs/by-name/ll/llama-cpp/package.nix

```diff
@@ -17,30 +16,19 @@
 , openclSupport ? false
 , clblast
 
-, openblasSupport ? !rocmSupport
+, openblasSupport ? !rocmSupport && !cudaSupport
 , openblas
 , pkg-config
 }:
 
-assert lib.assertMsg
-  (lib.count lib.id [openclSupport openblasSupport rocmSupport] == 1)
-  "llama-cpp: exactly one  of openclSupport, openblasSupport and rocmSupport should be enabled";
-
 let
-  cudatoolkit_joined = symlinkJoin {
-    name = "${cudaPackages.cudatoolkit.name}-merged";
-    paths = [
-      cudaPackages.cudatoolkit.lib
-      cudaPackages.cudatoolkit.out
-    ] ++ lib.optionals (lib.versionOlder cudaPackages.cudatoolkit.version "11") [
-      # for some reason some of the required libs are in the targets/x86_64-linux
-      # directory; not sure why but this works around it
-      "${cudaPackages.cudatoolkit}/targets/${stdenv.system}"
-    ];
-  };
-  metalSupport = stdenv.isDarwin && stdenv.isAarch64;
+  # It's necessary to consistently use backendStdenv when building with CUDA support,
+  # otherwise we get libstdc++ errors downstream.
+  effectiveStdenv = if cudaSupport then cudaPackages.backendStdenv else stdenv;
+  metalSupport = effectiveStdenv.isDarwin && effectiveStdenv.isAarch64 && !openclSupport;
 in
-stdenv.mkDerivation (finalAttrs: {
+# cuda only supports gcc11 at most for now.
```

Comment:
Btw this comment will be invalidated as soon as https://github.com/NixOS/nixpkgs/pull/269639 is merged. Better to clarify what "now" means, e.g. "cuda imposes an upper bound on the gcc version, e.g. the latest gcc compatible with cudaPackages_11 is gcc11"